packages/ember-toucan-core/src/components/form/fields/checkbox-group.hbs

```diff
@@ -16,7 +16,8 @@
       )
     }}
       <legend
-        class="type-md-tight text-body-and-labels flex items-center gap-1.5"
+        class="type-md-tight flex items-center gap-1.5
+          {{if @isDisabled 'text-disabled' 'text-body-and-labels'}}"
```

Comment:
It may be tempting to make a `GroupLegend` component or something, that's essentially `Label` but renders a `<legend` and can be used here and with the radio-group; however, I punted on this for now.  Happy to do it in a follow up if folks think it'd be worth it though.